webapp/updates.py

```diff
@@ -166,13 +171,21 @@ def _process_updates(self, packages_to_process, api_version, available_repo_ids,
                                                       valid_releasevers)
                     for repo_id in repo_ids:
                         repo_details = self.db_cache.repo_detail[repo_id]
-                        response['update_list'][pkg]['available_updates'].append({
-                            'package': nevra,
-                            'erratum': errata_name,
-                            'repository': repo_details[REPO_LABEL],
-                            'basearch': none2empty(repo_details[REPO_BASEARCH]),
-                            'releasever': none2empty(repo_details[REPO_RELEASEVER])
-                        })
+                        if api_version != 3:
+                            response['update_list'][pkg]['available_updates'].append({
+                                'package': nevra,
+                                'erratum': errata_name,
+                                'repository': repo_details[REPO_LABEL],
+                                'basearch': none2empty(repo_details[REPO_BASEARCH]),
+                                'releasever': none2empty(repo_details[REPO_RELEASEVER])
+                            })
+                        else:
+                            response['update_list'][pkg][nevra]['available_updates'].append({
```

Comment:
`pkg` is nevra of currently installed package, `nevra` is nevra of updated package, layering it like that and then accessing using `'available_updates'` is not optimal.

I was thinking more about something more structured like:

```
response['update_list'][pkg]['available_updates'] =
[
  {
    'package': nevra
    'errata': [
      {
        "name": "RHSA-2021-0001",
        "repositories": [{"label": label, "basearch": basearch, "releasever": releasever},]
      },
    ]
  },
]
```

also now, I'd rather make it as v4 API and not change v3

sorry for a very late review :slightly_frowning_face: